htdocs/compta/facture/card.php

```diff
@@ -3894,6 +3894,13 @@
 	if (empty($reshook)) $formconfirm .= $hookmanager->resPrint;
 	elseif ($reshook > 0) $formconfirm = $hookmanager->resPrint;
 
+	// recalculate remaining value because amounts might have changed in hook
```

Comment:
To avoid to reload payment e second time each time, can you try this change:
Add the variable   &$resteapayer  into list of parameters. Try to modify the value of it at end of trigger (by doing the $resteapayer = price2num($object->getRemainToPay(), 'MT') 
Did you succeed in modify the value this way ?

Reply:
Hello Laurent

Thank you for the hint! I tried it out and it works great for me. Indeed this might be the better solution.
The only thing I am wondering about is that you give the responsibility for the correct calculation for the remaining amount to pay into the hands of the person who uses the hook. 
However, if you think that this is better than recalling the function again - what in fact is also not a good solution - we should do it that way. This would result in the following changes:

The parameter's definition for the hook would change from
```
$parameters = array('lineid' => $lineid);
```
to
```
$parameters = array('lineid' => $lineid, 'remainingtopay' => &$resteapayer);
```
and there also has to be a repeated assignment (like somewhere some lines before), which I wasn't sure about what this is good for anyway:
```
if ($object->paye)
{
    $resteapayer = 0;
}
$resteapayeraffiche = $resteapayer;
```

Maybe the check for "if object paye" should be included in the getRemainToPay() function, too.

Reply:
Yes, i prefer having this value resteapayer provided as parameter and let the module modify it if he wants too.